torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,31 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        r"""
+        This is same as torch.optim.Optimizer state_dict,
+        but adds an extra entry to record model averager's step to the checkpoint
+        to ensure reload does not cause unnecessary warm up again.
+        """
+        optim_state_dict = self.optim.state_dict()
+        optim_state_dict['step'] = self.averager.step
+        return optim_state_dict
 
     def load_state_dict(self, state_dict):
+        r"""
+        This is same as torch.optim.Optimizer load_state_dict,
+        but also restores model averager's step value to the one,
+        saved in provided state_dict.
```

Comment:
(continued)

Here, I like to double backtick the `state_dict` to make it clear it is referring to the argument. This can be useful sometimes if, for example, the argument is `module`, and it is not necessarily clear from the context if you mean module conceptually or the argument `module`.
```suggestion
        saved in provided ``state_dict``.
```

Disclaimer: I think I am the only one who is picky about these docstring things. Nonetheless, it is still good to be aware of it.